test.py

```diff
@@ -64,6 +64,26 @@ def test_raises_for_non_csv(self, mock_urlopen):
         with self.assertRaisesRegexp(Exception, err):
             self.stream._api_call(None)
 
+    # Handles unencodable strings
+    @patch('ocr.urllib2.urlopen')
+    def test_hanldes_unencodable_strings(self, mock_urlopen):
+        res_mock = Mock()
+        res_mock.info.side_effect = self.foo
+        # The characters here represent a copywrite symbol
+        res_mock.read.side_effect = lambda: "{'problem_text': 'OHNO!\xc2\xae'}"
+        mock_urlopen.return_value = res_mock
+
+        exception_raised = False
+        try:
+            self.stream.read()
+        except Exception, e:
+            exception_raised = e
+
+        self.assertFalse(exception_raised, exception_raised)
```

Comment:
This test case is a specific test for the issue I'm trying to avoid with these changes. Within the files that were returned from the API, this content appeared - `\xc2\xae` - which we were unable to convert to `utf8`. This caused an Exception to be raised when writing to to temp file.

So this test ensures that the "problematic" data is returned but doesn't raise an exception. I don't think that `raise Exception('')` is a feasible scenario here since the errors that are raised are python internal and not custom errors.